tenets/codelingo/go/unused-private-const-var-same-package/codelingo.yaml

```diff
@@ -0,0 +1,29 @@
+tenets:
+  - name: unused-private-const-var-same-package
+    doc: |
+      Find private const or var which are unused in a given package.
+    flows:
+      codelingo/review:
+        comment: |
+          Private global {{globalname}} is never used.  It should probably be removed.
+    query: |
+      import codelingo/ast/go
+
+      go.dir:
+        go.file(depth = any):
+          go.decls:
+            go.gen_decl:
+              go.value_spec:
+                go.names:
+                  @ review.comment
+                  go.ident:
+                    @ review.vars.globalname
+                    name as privateVarName
+                    private == "true"
+        exclude:
+          go.file(depth = any):
+            go.call_expr(depth = any):
```

Comment:
Does it matter that the var/const is used in a function call? If you changed the `foo` function to:
```
func foo() {
    a := privateVarA
    fmt.Printf("%s\n", a)
}
```

I suspect your tenet would suddenly think `privateVarA` was unused.